telemetry/prometheus/src/main/scala/io/buoyant/telemetry/prometheus/PrometheusTelemeter.scala

```diff
@@ -75,14 +82,21 @@ class PrometheusTelemeter(metrics: MetricsTree, private[prometheus] val handlerP
     val (prefix1, labels1) = prefix0 match {
       case Seq("rt", router) if !labelExists(labels0, "rt") =>
         (Seq("rt"), labels0 :+ ("rt" -> escapeLabelVal(router)))
-      case Seq("rt", "service", path) if !labelExists(labels0, "service") =>
-        (Seq("rt", "service"), labels0 :+ ("service" -> escapeLabelVal(path)))
-      case Seq("rt", "client", id) if !labelExists(labels0, "client") =>
-        (Seq("rt", "client"), labels0 :+ ("client" -> escapeLabelVal(id)))
+
+      // Add label for stack { "service", "client", "server" }
+      case Seq("rt", stack, identifier) if !labelExists(labels0, stack) =>
+        (Seq("rt", stack), labels0 :+ (stack -> escapeLabelVal(identifier)))
+
+      // Handle client service case
       case Seq("rt", "client", "service", path) if !labelExists(labels0, "service") =>
         (Seq("rt", "client", "service"), labels0 :+ ("service" -> escapeLabelVal(path)))
-      case Seq("rt", "server", srv) if !labelExists(labels0, "server") =>
-        (Seq("rt", "server"), labels0 :+ ("server" -> escapeLabelVal(srv)))
+
+      // Add label for exception { "failures", "exn" }
+      case Seq("rt", stack, "failures", exception) =>
+        (Seq("rt", stack, "failures"), addException(labels0, exception))
```

Comment:
I think this could be more simply written as something like:

```
  case Seq("rt", stack, "failures", exception@_*) => (..., labels0 :+ ("exception" -> escapeLabelVal(exception.mkString(":")))))
```
```

Reply:
That's what I was originally working with, but a nested exception is built incrementally - not all at once. In the case of `.counter("bar", "qux")`, the first pass through `writeMetrics` will only see `bar`, and the second pass will only see `qux`.

We want to append `qux` to the already existing exception `bar`, but we can't do that in one go.